javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -809,13 +809,13 @@ public CharSeq reverse() {
     @Override
     public <U> IndexedSeq<U> scanLeft(U zero, BiFunction<? super U, ? super Character, ? extends U> operation) {
         Objects.requireNonNull(operation, "operation is null");
-        return Collections.scanLeft(this, zero, operation, Vector.empty(), Vector::append, Function.identity());
+        return Collections.scanLeft(this, zero, operation, Iterator::toVector);
```

Comment:
I've recognized one thing that is outside (but related) to this PR: CharSeq.scan() currently returns IndexedSeq&lt;Character&gt; but it should return CharSeq.

Could you please change the return type of scan() to CharSeq?

```java
@Override
public IndexedSeq<Character> scan(Character zero, BiFunction<? super Character, ? super Character, ? extends Character> operation) {
    return Collections.scanLeft(this, zero, operation, Iterator::toCharSeq);
}
```